### PR TITLE
Catapult backend

### PR DESCRIPTION
**Pull request by dgburnette**
# Description

Added safeguard for deref of non-existent attribute.

- [ ] Bug fix (non-breaking change that fixes an issue)

No test was added

> :memo: Please describe the tests that you ran to verify your changes.
>
> * Provide instructions so we can reproduce.
> * Please also list any relevant details for your test configuration.

**Test Configuration**:

## Checklist

- [ ] I have read the [guidelines for contributing](https://github.com/fastmachinelearning/hls4ml/blob/main/CONTRIBUTING.md).
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have installed and run `pre-commit` on the files I edited or added.
- [ ] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: hls4ml/model/profiling.py [hunk 1]
```diff
@@ -589,6 +589,7 @@ def get_ymodel_keras(keras_model, X):
         name = layer.name
         if (
             hasattr(layer, 'activation')
+            and hasattr(layer.activation, "__name__")
```
Comment (jmitrevs) [inline hls4ml/model/profiling.py:592]: Is this really needed? I think there were two parallel fixes for this in two pull requests, and all layer.acitvations that are of type `Activation` or `QActivation` should have __name__ defined, right?

**Comment by dgburnette** [comment]
Hi Jovan,
You may be right. This was something we (Giuseppe and I) stumbled across a month or so ago.

**Comment by jmitrevs** [comment]
I am thinking of closing this, unless you run into the problem again. I believe the addition of the check on the type is sufficient. It fixed the problem when I ran into it.

**Comment by jmitrevs** [comment]
Please reopen it if the problem does not go away.

**Comment by dgburnette** [comment]
Ok. That is fine with me.

**Pull request closed without merging**